ot/mpz-ot/src/actor/kos/receiver.rs

```diff
@@ -0,0 +1,346 @@
+use std::collections::{HashMap, VecDeque};
+
+use async_trait::async_trait;
+use futures::{
+    channel::{mpsc, oneshot},
+    stream::Fuse,
+    SinkExt, StreamExt,
+};
+use utils_aio::{
+    non_blocking_backend::{Backend, NonBlockingBackend},
+    sink::IoSink,
+    stream::IoStream,
+};
+
+use crate::{
+    kos::{Receiver, ReceiverError, ReceiverKeys},
+    OTError, OTReceiverShared, OTSetup, VerifiableOTReceiverShared, VerifiableOTSender,
+};
+use mpz_core::{Block, ProtocolMessage};
+use mpz_ot_core::kos::{msgs::SenderPayload, PayloadRecord};
+
+use crate::actor::kos::{
+    into_kos_sink, into_kos_stream,
+    msgs::{ActorMessage, Message, TransferPayload, TransferRequest},
+    ReceiverActorError,
+};
+
+enum Command {
+    Receive {
+        id: String,
+        choices: Vec<bool>,
+        caller_response: oneshot::Sender<Result<(ReceiverKeys, SenderPayload), ReceiverError>>,
+    },
+    Verify(Verify),
+    Shutdown(Shutdown),
+}
+
+struct Verify {
+    id: String,
+    caller_response: oneshot::Sender<Result<PayloadRecord, ReceiverError>>,
+}
+
```

Comment:
/// A pending oblivious transfer which was requested by this KOS receiver but has not yet been
/// responded to by the KOS sender.